sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -237,6 +239,38 @@ case class CatalogTable(
 }
 
 
+/**
+ * This class of statistics is used in [[CatalogTable]] to interact with metastore.
+ */
+case class CatalogStatistics(
+    sizeInBytes: BigInt,
+    rowCount: Option[BigInt] = None,
+    colStats: Map[String, ColumnStat] = Map.empty) {
+
+  /**
+   * Convert [[CatalogStatistics]] to [[Statistics]], and match column stats to attributes based
+   * on column names.
+   */
+  def convert(attributes: Seq[Attribute]): Statistics = {
+    val matched = mutable.HashMap[Attribute, ColumnStat]()
+    attributes.foreach { attr =>
+      if (colStats.contains(attr.name)) {
+        matched.put(attr, colStats(attr.name))
+      }
+    }
+    Statistics(sizeInBytes = sizeInBytes, rowCount = rowCount,
+      attributeStats = AttributeMap(matched.toSeq))
+  }
+
+  /** Readable string representation for the CatalogStatistics. */
+  def simpleString: String = {
```

Comment:
Because we don't print column stats in it, it's not a "complete" string representation. Column stats can be too much and make CatalogTable unreadable.